From c72deffcfad3c7d0045c2668c340f2beb4eb0b3c Mon Sep 17 00:00:00 2001
From: Hongcheng Xie <hongcheng.xie@intel.com>
Date: Fri, 4 Sep 2020 10:53:07 +0800
Subject: [PATCH] Don't block to wait for BPF loader

Temp don't block and always waiting for BPF Loader ready
since BPF Loader failed to run now.

Change-Id: Ib485e914c516e8ef576fcd4ab0900c899f7a64d4
Tracked-On:
Signed-off-by: Hongcheng Xie <hongcheng.xie@intel.com>
---
 libbpf_android/Loader.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/libbpf_android/Loader.cpp b/libbpf_android/Loader.cpp
index da6a109..dfe85d6 100644
--- a/libbpf_android/Loader.cpp
+++ b/libbpf_android/Loader.cpp
@@ -674,7 +674,8 @@ static bool waitSecondsForProgsLoaded(int seconds) {
 
 void waitForProgsLoaded() {
     if (!android::bpf::isBpfSupported()) return;
-
+    ALOGW("Temp return directly to avoid netd blocking");
+    return;
     if (waitSecondsForProgsLoaded(5)) return;
     if (waitSecondsForProgsLoaded(10)) return;
     if (waitSecondsForProgsLoaded(20)) return;
-- 
2.17.1

